components/gitpod-db/src/typeorm/migration/1673944879912-AddOrganisationIDToOIDCClientConfig.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { columnExists, indexExists } from "./helper/helper";
+
+const table = "d_b_oidc_client_config";
+const column = "organisationId";
+const orgIndex = "idx_organisationId";
+const idOrgCompositeIndex = "idx_id_organisationId";
+
+export class AddOrganisationIDToOIDCClientConfig1673944879912 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        if (!(await columnExists(queryRunner, table, column))) {
+            await queryRunner.query(
+                `ALTER TABLE ${table} ADD COLUMN ${column} varchar(255), ALGORITHM=INPLACE, LOCK=NONE`,
+            );
+        }
+
+        if (!(await indexExists(queryRunner, table, orgIndex))) {
+            await queryRunner.query(`CREATE INDEX \`${orgIndex}\` ON \`${table}\` (${column})`);
+        }
+
+        if (!(await indexExists(queryRunner, table, idOrgCompositeIndex))) {
+            await queryRunner.query(`CREATE INDEX \`${idOrgCompositeIndex}\` ON \`${table}\` (id, ${column})`);
```

Comment:
💡 Idea, I assume `idx_id_organisationId` is optimizing for queries where we'd use `id` and `orgId` to fetch a single entry.

`id` should be already unique within the system. When trying to read the client config from DB we should already have the `id` at hand, which can be used for direct access. It's not obvious to see the benefit of `idx_id_organisationId`.

Reply:
The reason is the same as in https://github.com/gitpod-io/gitpod/pull/15777#discussion_r1072246207, the lookup should be by the tuple (id, org_id).

Reply:
@AlexTugarev Do you want me to drop the extra index?